### DISCUSSION TITLE
Is it possible to call Widgets' function from backend code, like rust?

### DISCUSSION THREAD

**Discussion by FischLu** [slint-ui/slint#6343] [Q&A] [answered]
In Slint, many widgets come with built-in functions, such as the `select-all()` function in `LineEdit`. I can call this function within the .slint file. My question is: is it possible to invoke such a function from back end code like Rust? For example, can I call `select-all()` on a specific `LineEdit` via its ID/name when a condition in Rust is met? If so, how can I achieve this? Thanks in advance!

**Comment by Enyium** [comment, marked as answer]
Only by calling through to it:

```slint
export component AppWindow inherits Window {
    public function select-all-line-edit-text() { my-line-edit.select-all(); }
    //...
}
```

Then call `app_window.invoke_select_all_line_edit_text()` from Rust.

**Reply by FischLu** [reply]
Nice solution, it works, thank you!
